### ISSUE TITLE
Error in installation instructions

### ISSUE THREAD

**Issue by wincowgerDEV** [beniaminogreen/zoomerjoin#70]
**Describe the bug**
I am running Windows 11 with rust 1.71.1 and wsl, when I try to install the package this is my console with an error message:

```r
> library(zoomerjoin)
Error in library(zoomerjoin) : there is no package called ‘zoomerjoin’
> ## Install with devtools
> # install.packages("devtools")
> devtools::install_github("beniaminogreen/zoomerjoin")
Downloading GitHub repo beniaminogreen/zoomerjoin@HEAD
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

1: All                          
2: CRAN packages only           
3: None                         
4: vctrs (0.6.2 -> 0.6.3) [CRAN]

Enter one or more numbers, or an empty line to skip updates: 3
── R CMD build ───────────────────────────────────────────────────────────────────────────────────────────────────────────
✔  checking for file 'C:\Users\winco\AppData\Local\Temp\RtmpCwBEaE\remotes522c7688750f\beniaminogreen-zoomerjoin-f359f28/DESCRIPTION' ...
─  preparing 'zoomerjoin':
✔  checking DESCRIPTION meta-information ... 
─  cleaning src
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building 'zoomerjoin_0.0.0.9000.tar.gz'
   
Installing package into ‘C:/Users/winco/AppData/Local/R/win-library/4.3’
(as ‘lib’ is unspecified)
* installing *source* package 'zoomerjoin' ...
** using staged installation
** libs
using C compiler: 'gcc.exe (GCC) 12.2.0'
rm -Rf zoomerjoin.dll ./rust/target/x86_64-pc-windows-gnu/release/libzoomerjoin.a entrypoint.o
gcc  -I"C:/PROGRA~1/R/R-43~1.0/include" -DNDEBUG     -I"C:/RBuildTools/4.3/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -c entrypoint.c -o entrypoint.o
mkdir -p ./rust/target/libgcc_mock
cd ./rust/target/libgcc_mock && \
	touch gcc_mock.c && \
	gcc -c gcc_mock.c -o gcc_mock.o && \
	ar -r libgcc_eh.a gcc_mock.o && \
	cp libgcc_eh.a libgcc_s.a
C:\RBuildTools\4.3\x86_64-w64-mingw32.static.posix\bin\ar.exe: creating libgcc_eh.a
# CARGO_LINKER is provided in Makevars.ucrt for R >= 4.2
export CARGO_TARGET_X86_64_PC_WINDOWS_GNU_LINKER="x86_64-w64-mingw32.static.posix-gcc.exe" && \
	export LIBRARY_PATH="${LIBRARY_PATH};/c/Users/winco/AppData/Local/Temp/Rtmpctc8EF/R.INSTALL1c04338dae1/zoomerjoin/src/./rust/target/libgcc_mock" && \
	cargo +stable-gnu build --target=x86_64-pc-windows-gnu --lib --release --manifest-path=./rust/Cargo.toml --target-dir ./rust/target
error: toolchain 'stable-x86_64-pc-windows-gnu' is not installed
make: *** [Makevars.win:19: rust/target/x86_64-pc-windows-gnu/release/libzoomerjoin.a] Error 1
ERROR: compilation failed for package 'zoomerjoin'
* removing 'C:/Users/winco/AppData/Local/R/win-library/4.3/zoomerjoin'
Warning message:
In i.p(...) :
  installation of package ‘C:/Users/winco/AppData/Local/Temp/RtmpCwBEaE/file522c5d5356f/zoomerjoin_0.0.0.9000.tar.gz’ had non-zero exit status
```

I found this guidance https://github.com/rust-lang/rustup/issues/1793 and tried to follow along but the package is still not installing. 

This is the result of rustup show for me:

```
Default host: x86_64-unknown-linux-gnu
rustup home:  /home/wincowger/.rustup

stable-x86_64-unknown-linux-gnu (default)
rustc 1.71.1 (eb26296b5 2023-08-03)
```

**To Reproduce**
Follow user guidelines on Windows 11 machine with WSL. 

**Expected behavior**
Installation to proceed. 

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 11

**Additional context**

**Comment by beniaminogreen** [comment]
Thanks for filing this! I do not have experience installing the package on a Windows computer with WSL, but I have had to complete extra steps when installing on a Windows server I have access to.  Generally, I have to make sure that the correct rust toolchains are installed, which doesn't seem to be the case here (although I don't have much experience using Windows). 

I don't mean to suggest that you run something you've already tried, but can you try running the following commands?

```
rustup toolchain remove stable-x86_64-pc-windows-msvc
rustup install stable-x86_64-pc-windows-msvc
rustup toolchain remove stable-x86_64-pc-windows-gnu
rustup install stable-x86_64-pc-windows-gnu
```

Here's the output of my rustup-show on the windows server I have access to: 

```
rustup show
Default host: x86_64-pc-windows-msvc
rustup home:  C:\Users\bfg9\.rustup

installed toolchains
--------------------

stable-x86_64-pc-windows-gnu
stable-x86_64-pc-windows-msvc (default)
nightly-x86_64-pc-windows-msvc

installed targets for active toolchain
--------------------------------------

i686-pc-windows-gnu
x86_64-pc-windows-gnu
x86_64-pc-windows-msvc

active toolchain
----------------

stable-x86_64-pc-windows-msvc (default)
rustc 1.70.0 (90c541806 2023-05-31)
```

I can see Windows installation as being a stumbling block for a lot of users, so I will make sure to update the instructions in the README after I discover a reproducible fix. 

Best, Ben

**Comment by wincowgerDEV** [comment]
Thanks for this. Definitely getting closer, may need to change the active toolchain.:

```
wincowger@Win:/mnt/c/WINDOWS/system32$ rustup install stable-x86_64-pc-windows-msvc
error: DEPRECATED: future versions of rustup will require --force-non-host to install a non-host toolchain as the default.
warning: toolchain 'stable-x86_64-pc-windows-msvc' may not be able to run on this system.
warning: If you meant to build software to target that platform, perhaps try `rustup target add x86_64-pc-windows-msvc` instead?
info: syncing channel updates for 'stable-x86_64-pc-windows-msvc'
info: latest update on 2023-08-03, rust version 1.71.1 (eb26296b5 2023-08-03)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'rust-docs'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: installing component 'clippy'
info: installing component 'rust-docs'
 13.6 MiB /  13.6 MiB (100 %)  10.5 MiB/s in  1s ETA:  0s
info: installing component 'rust-std'
 23.6 MiB /  23.6 MiB (100 %)  18.7 MiB/s in  1s ETA:  0s
info: installing component 'rustc'
 59.5 MiB /  59.5 MiB (100 %)  20.3 MiB/s in  2s ETA:  0s
info: installing component 'rustfmt'

  stable-x86_64-pc-windows-msvc installed - (rustc does not exist)

info: checking for self-update
wincowger@Win:/mnt/c/WINDOWS/system32$ rustup toolchain remove stable-x86_64-pc-windows-gnu
info: no toolchain installed for 'stable-x86_64-pc-windows-gnu'
wincowger@Win:/mnt/c/WINDOWS/system32$ rustup install stable-x86_64-pc-windows-gnu
error: DEPRECATED: future versions of rustup will require --force-non-host to install a non-host toolchain as the default.
warning: toolchain 'stable-x86_64-pc-windows-gnu' may not be able to run on this system.
warning: If you meant to build software to target that platform, perhaps try `rustup target add x86_64-pc-windows-gnu` instead?
info: syncing channel updates for 'stable-x86_64-pc-windows-gnu'
info: latest update on 2023-08-03, rust version 1.71.1 (eb26296b5 2023-08-03)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'rust-docs'
 13.6 MiB /  13.6 MiB (100 %)   6.0 MiB/s in  2s ETA:  0s
info: downloading component 'rust-mingw'
info: downloading component 'rust-std'
info: downloading component 'rustc'
 71.4 MiB /  71.4 MiB (100 %)  17.5 MiB/s in  4s ETA:  0s
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: installing component 'clippy'
info: installing component 'rust-docs'
 13.6 MiB /  13.6 MiB (100 %)  13.0 MiB/s in  1s ETA:  0s
info: installing component 'rust-mingw'
info: installing component 'rust-std'
 23.2 MiB /  23.2 MiB (100 %)  20.1 MiB/s in  1s ETA:  0s
info: installing component 'rustc'
 71.4 MiB /  71.4 MiB (100 %)  22.3 MiB/s in  3s ETA:  0s
info: installing component 'rustfmt'

  stable-x86_64-pc-windows-gnu installed - (rustc does not exist)

info: checking for self-update
wincowger@Win:/mnt/c/WINDOWS/system32$ rustup show
Default host: x86_64-unknown-linux-gnu
rustup home:  /home/wincowger/.rustup

installed toolchains
--------------------

stable-x86_64-pc-windows-gnu
stable-x86_64-pc-windows-msvc
stable-x86_64-unknown-linux-gnu (default)

active toolchain
----------------

stable-x86_64-unknown-linux-gnu (default)
rustc 1.71.1 (eb26296b5 2023-08-03)
```

**Comment by beniaminogreen** [comment]
Hmm. Seems like you are very close. If it's not working already, you could try: 
```
rustup default stable-x86_64-pc-windows-gnu
```

To make sure that the x86-windows-gnu toolchain is set as the default toolchain

**Comment by wincowgerDEV** [comment]
Ok, I have the default toolchain set. 
```
rustup show
Default host: x86_64-unknown-linux-gnu
rustup home:  /home/wincowger/.rustup

installed toolchains
--------------------

stable-x86_64-pc-windows-gnu (default)
stable-x86_64-pc-windows-msvc
stable-x86_64-unknown-linux-gnu

active toolchain
----------------

stable-x86_64-pc-windows-gnu (default)
(rustc does not exist)
```

However, I am still getting the same error message when I try to install the package. I did realize there may not be as much connection between WSL and Windows as I previously thought so I tried to run RStudio from WSL using this example: https://support.posit.co/hc/en-us/articles/360049776974-Using-RStudio-Server-in-Windows-WSL2, but that is now giving me a bunch of issues too, lots of errors that don't have resolutions when I check online. What is the workflow you usually use to install on a windows machine?

**Comment by beniaminogreen** [comment]
Shoot, sorry you are still having issues with this!

Just installed Zoomerjoin on my partner's windows computer, which worked after performing the following steps (which are also what I have done in the past). 

1) Install rustup-init.exe from [this site](https://forge.rust-lang.org/infra/other-installation-methods.html) and run it
2) run `rustup install stable-x86_64-pc_windows-gnu` to install the correct toolchain
3) Install zoomerjoin with `devtools::install_github("beniaminogreen/zoomerjoin")`

Reading that article it looks as if Rstudio might not be supported for WSL, which could make the install more difficult (although I am not sure this is the case). I just want to confirm that trying to install both Rust and the R package in WSL, or trying to install both in Windows, correct? I don't think that you would be able to install Rust in WSL and the package with Windows and have the two work together, but I am not sure. 

I'll try and look into how the install works on WSL, but I am not sure that I will have access to a device with it on anytime soon. 

Best, 
Ben

**Comment by wincowgerDEV** [comment]
Great! Got it working. Here is what I did. I think my main issue was that I tried to use WSL, it wasn't necessary in this case. 

1. go to https://forge.rust-lang.org/infra/other-installation-methods.html, install from this link 
![image](https://github.com/beniaminogreen/zoomerjoin/assets/26821843/8feacef5-c557-44c4-a207-f2b2a175721f)
2. in command prompt run rustup install ```stable-x86_64-pc-windows-msvc``` then run ```rustup install stable-x86_64-pc-windows-gnu``` then run ```rustup default stable-x86_64-pc-windows-gnu```. 
3. Install zoomerjoin with ```devtools::install_github("beniaminogreen/zoomerjoin")``` in R.
